### PR TITLE
NIFI-8779 correct Docker build scripts

### PR DESCRIPTION
**Pull request by ChrisSamo632**
#### Description of PR

NIFI-8779 correctly pass MIRROR_BASE_URL from DockerBuild to Dockerfile; address shellcheck issues with Docker-related scripts

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- ~[ ] Have you written or updated unit tests to verify your changes?~
- [x] Have you verified that the full build is successful on JDK 8?
- [x] Have you verified that the full build is successful on JDK 11?
- ~[ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?~
- ~[ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?~
- ~[ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?~
- ~[ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?~

### For documentation related changes:
- ~[ ] Have you ensured that format looks appropriate for the output in which it is rendered?~

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-docker/dockerhub/DockerBuild.sh [hunk 1]
```diff
@@ -13,24 +15,16 @@
...
+set -o pipefail
 
-DOCKER_GID=1000
-if [ -n "$2" ]; then
-  DOCKER_GID="$2"
-fi
+DOCKER_UID="${1:-1000}"
+DOCKER_GID="${2:-1000}"
+MIRROR="${3:-https://archive.apache.org/dist}"
+BASE="${4:-${MIRROR}}"
```
Comment (jfrazee) [inline nifi-docker/dockerhub/DockerBuild.sh:24]: Can I suggest we do the mirror handling for the user (either here or in the Dockerfile itself)? For example, you can determine whether to go to archive.a.o or not by checking the in_dist from the Apache closer.lua script:

```sh
curl -sSLo - "https://www.apache.org/dyn/closer.lua/nifi/${NIFI_VERSION}/nifi-${NIFI_VERSION}-bin.zip?as_json&cca2=us" | \
    jq -r '(if .in_dist then .preferred else "https://archive.apache.org/dist/" end) + .path_info'
```

Comment (ChrisSamo632) [reply nifi-docker/dockerhub/DockerBuild.sh:24]: That adds a jq dependency on the builder's machine (unless we do that in the Dockerfile, although that means installing the dependencies before setting some of the ARG/ENV, and that feels wrong)

Also, in_dist appears to always be false (chewing 1.13.2, 1.14.p, 1.15.0 and 1.15.0-SNAPSHOT), so we'll always build from the default archive site, which won't contain the in-progress RC artifacts. Unless I'm missing something (I've got no idea what the link is that you've referenced, so don't know what creates/updates it, etc.)

File: nifi-docker/dockerhub/pom.xml [hunk 2]
```diff
@@ -24,17 +24,26 @@
...
     </properties>
 
     <profiles>
         <profile>
             <id>docker</id>
+            <dependencies>
+                <dependency>
+                    <groupId>javax.activation</groupId>
+                    <artifactId>activation</artifactId>
+                    <version>1.1.1</version>
```
Comment (exceptionfactory) [inline nifi-docker/dockerhub/pom.xml:39]: Is there a particular reason for adding this dependency?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for improving the Docker scripts @ChrisSamo632! Most of the changes appear straightforward, I just noted one question on the addition of the `activation` dependency in the Docker modules.

**Comment by kevdoran** [comment]
Thanks for this @ChrisSamo632. I should get to review it this afternoon

**Comment by kevdoran** [comment]
I've been testing this, and I'm having trouble getting the dockermaven image working correctly (dockerhub seems to be working fine). NiFi fails right when initializing the core JettyServer instance:

```
nifi_1      | 2021-11-22 14:55:59,786 INFO [main] org.apache.nifi.nar.NarClassLoaders Found NiFiServer implementation org.apache.nifi.web.server.JettyServer in org.apache.nifi:nifi-server-nar:1.16.0-SNAPSHOT
nifi_1      | 2021-11-22 14:55:59,819 INFO [main] org.eclipse.jetty.util.log Logging initialized @35088ms to org.eclipse.jetty.util.log.Slf4jLog
nifi_1      | 2021-11-22 14:55:59,914 INFO [main] org.apache.nifi.web.server.JettyServer Configuring Jetty for HTTPS on port: 8443
nifi_1 exited with code 0
```

I'm using the same config for both (a docker compose file):

```
version: "3.5"

services:

  registry:
    image: apache/nifi-registry:1.16.0-SNAPSHOT-dockermaven
    ports:
      - "18080:18080" # UI
    networks:
      - demo

  nifi:
    image: apache/nifi:1.16.0-SNAPSHOT-dockermaven
    ports:
      - "8443:8443" # UI
      - "10000"
    networks:
      - demo
    environment:
      SINGLE_USER_CREDENTIALS_USERNAME: admin
      SINGLE_USER_CREDENTIALS_PASSWORD: admin-password
      NIFI_SENSITIVE_PROPS_KEY: notAGoodPassword

networks:
  demo: 
```

I'm going to spend some time today looking into it. It could be something in my local dev environment causing the problem. Just wanted to provide an update, will share more later today.

**Review by kevdoran — APPROVED** [review]
After doing a clean build, my issues with the local dockermaven image were resolved. All looks good on my end. Thanks for the contribution @ChrisSamo632! +1, will merge

**Pull request closed without merging**